### DISCUSSION TITLE
Ouput 'console.log();' doesn't go to the right console!? help me :)

### DISCUSSION THREAD

**Discussion by Alama32** [microsoft/vscode-discussions#971] [Extension Development QnA] [answered]
Hello everyone, I am trying to code an extension for vscode in JS. So far, it seems to work well except for one detail.
When I press F5 it opens a Development Host window, it's good so far, but surprisingly, the output to the console requested by 'console.log(""); ' does not come out in this Host window but in the editor console., as this Host window is positioned on top, I wasted a lot of time thinking that my program was never executed even though everything seemed correct .
Do you have any idea why my outputs are not outputting to the correct console?
Thank you in advance for your answers :-)

Here is my launch.json
```json
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Launch Extension",
			"type": "extensionHost",
			"request": "launch",
			"runtimeExecutable": "${execPath}",
			"args": [
				"--extensionDevelopmentPath=${workspaceRoot}"
			]
		}
	]
}
```

**Comment by r3m0t** [comment, marked as answer]
Hi, it's expected to go to the Debug Console in the main window (not the temporary one that appears when you press F5).

You can use the `createOutputChannel` API and use the Output panel in the Extension Development Host window, and your users will appreciate you logging there as well.

**Comment by Alama32** [comment]
Thank for your answer  :)
